sympy/sets/tests/test_sets.py

```diff
@@ -1363,6 +1363,16 @@ def test_issue_11827():
     assert S.Naturals0**4
 
 
+def test_issue_14336():
+    try:
+        U = S.Complexes
+        x = Symbol("x")
+        U -= U.intersect(Ne(x, 1).as_set())
+        U -= U.intersect(S.true.as_set())
+    except RecursionError:
+        assert False
```

Comment:
We don't need to catch the error just to assert False. The exception will cause the test to fail anyway.